x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -1214,13 +1215,14 @@ private static void renderTypes(List<String> argNames) throws IOException {
 
         List<String> table = new ArrayList<>();
         for (Map.Entry<List<DataType>, DataType> sig : signatures().entrySet()) { // TODO flip to using sortedSignatures
-            if (sig.getKey().size() != argNames.size()) {
+            if (sig.getKey().size() > argNames.size()) { // skip variadic [test] cases (but not those with optional parameters)
```

Comment:
There is a variadic flag in the FunctionDescription, that we can check if a function is variadic, has a List as input argument. Some functions have optional argument, but not necessarily variadic, like bucket.

Reply:
Right, the FunctionDescription#variadic record component is [currently only true](https://github.com/elastic/elasticsearch/blob/f218c7b4996abb8e2747a8dacdb3575056ac9fb9/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java#L402) if there's at least one parameter that's an Expression list (which usually should be the last one, not sure if we can safely resolve the types otherwise).
`BUCKET` isn't really variadic, it just supports this overloaded invocation. That's why the check doesn't make use of `variadic`.